zutil.c

```diff
@@ -102,37 +103,12 @@ const char * Z_EXPORT PREFIX(zError)(int err) {
     return ERR_MSG(err);
 }
 
-#ifndef MY_ZCALLOC /* Any system without a special alloc function */
-
-#ifndef UNALIGNED_OK
-#  include <malloc.h>
-#  if defined(_WIN32)
-#    define zng_align_alloc(align, size) _aligned_malloc(size, align)
-#    define zng_align_free(ptr)          _aligned_free(ptr)
-#  else
-#    define zng_align_alloc              memalign
-#    define zng_align_free(ptr)          free(ptr)
-#  endif
-#endif
-
-void Z_INTERNAL *zng_calloc(void *opaque, unsigned items, unsigned size)
-{
+void Z_INTERNAL *zng_calloc(void *opaque, unsigned items, unsigned size) {
     (void)opaque;
-#ifndef UNALIGNED_OK
-    return zng_align_alloc(16, items * size);
-#else
-    return sizeof(unsigned int) > 2 ? (void *)malloc(items * size) :
-                              (void *)calloc(items, size);
-#endif
+    return zng_alloc(items * size);
```

Comment:
Multiplying without widening is unsafe, should cast both to size_t to make sure it can't overflow when we made the function public.

Reply:
Fixed